remote-messaging/remote-messaging-impl/src/main/java/com/duckduckgo/remote/messaging/impl/mappers/JsonRemoteMessageMapper.kt

```diff
@@ -100,11 +99,17 @@ private val defaultBrowserActionMapper: (JsonMessageAction) -> Action = {
     Action.DefaultBrowser()
 }
 
+private val appTpOnboardingActionMapper: (JsonMessageAction) -> Action = {
+    Action.AppTpOnboarding()
+}
+
+// plugin point?
 private val actionMappers = mapOf(
     Pair(URL.jsonValue, urlActionMapper),
     Pair(DISMISS.jsonValue, dismissActionMapper),
     Pair(PLAYSTORE.jsonValue, playStoreActionMapper),
     Pair(DEFAULT_BROWSER.jsonValue, defaultBrowserActionMapper),
+    Pair(APP_TP_ONBOARDING.jsonValue, appTpOnboardingActionMapper),
```

Comment:
Yeah, that was what I wanted to do initially. But then I struggle to turn this mappers into plugins without heavier refactors. Because atm they're all called from either extension functions, or global objects etc.

That's why I left the `plugin point?` comment, so that we can come back to this in future and do that refactor

Reply:
👍 

FYI: Gave this a try last night. While moving to plugins was not an issue if we inject them in the root method that triggers the mapping...I hit an issue when trying to create an `Action` that has an `Intent` as an attribute. When we are done parsing and evaluating the messages, we persist the winner message in the database as a `Json`. While there are no issues serializing/deserializing into json the current actions, we need a workaround to serialize/deserialize an action that has an Intent.